plugins/reboot/reboot.go

```diff
@@ -50,24 +102,51 @@ func (plugin) NewTaskPlugin(options plugins.TaskPluginOptions) (plugins.TaskPlug
 		return nil, engines.ErrContractViolation
 	}
 
+	pl.runningTasks.RLock()
 	return taskPlugin{
 		TaskPluginBase: plugins.TaskPluginBase{},
-		reboot:         p.Reboot,
+		plugin:         pl,
+		rebootWhenDone: p.RebootWhenDone,
 	}, nil
 }
 
 func (tp taskPlugin) Dispose() error {
-	if tp.reboot {
-		if err := reboot(); err != nil {
-			return engines.NewInternalError(err.Error())
-		}
+	tp.plugin.runningTasks.RUnlock()
+
+	// Note: in a multi-task environment, this will not reboot immediately, but
+	// will wait until all tasks are complete.
+	if tp.rebootWhenDone {
+		tp.plugin.environment.Log.Warning("Task specified reboot after completion; scheduling reboot")
+		tp.plugin.reboot()
 	}
 
 	return nil
 }
 
 func (pluginProvider) NewPlugin(options plugins.PluginOptions) (plugins.Plugin, error) {
-	return plugin{}, nil
+	var c config
+	if err := schematypes.MustMap(configSchema, options.Config, &c); err != nil {
+		return nil, engines.ErrContractViolation
+	}
+
+	plugin := &plugin{
+		PluginBase:   plugins.PluginBase{},
+		environment:  options.Environment,
+		runningTasks: sync.RWMutex{},
+	}
+
+	if c.RebootAfter > 0 {
```

Comment:
Given that this is so far out into the future... why not just set `rebootAfter` as a `time.Time` timestamp...
Then do `if time.Now().Before(tp.plugin.rebootAfter) {  reboot(); }` we probably don't need a timer when we're waiting 96 hours... or so...